components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -6764,9 +6813,16 @@ public void deleteAPIProduct(APIProductIdentifier identifier) throws APIManageme
         //this is the product rxt instance path
         String apiProductArtifactPath = APIUtil.getAPIProductPath(identifier);
 
-        //todo : check whether there are any subscriptions for this api
-
         try {
+            //int apiId = apiMgtDAO.getAPIID(identifier, null);
+            long subsCount = apiMgtDAO.getAPISubscriptionCountByAPI(identifier);
+            if (subsCount > 0) {
+                //Logging as a WARN since this isn't an error scenario.
+                String message = "Cannot remove the API Product as active subscriptions exist.";
+                log.warn(message);
+                throw new APIManagementException(message);
```

Comment:
A small suggestion: We can set an ExceptionCode to easily send a proper response to the client. 
https://github.com/wso2/carbon-apimgt/wiki/Exception-handling-in-REST-APIs#procedure

Reply:
This check is already handled at Rest API level in ApiProductsApiServiceImpl.apiProductsApiProductIdDelete() method. This validation is already included in the PR